src/core/functions/scalar/llm_filter.cpp

```diff
@@ -107,13 +109,28 @@ static void LlmFilterScalarFunction2(DataChunk &args, ExpressionState &state, Ve
     Connection con(*state.GetContext().db);
     CoreScalarParsers::LlmFilterScalarParser(args);
 
+    nlohmann::json settings;
+    if (args.ColumnCount() == 4) {
+        settings = CoreScalarParsers::Struct2Json(args.data[3], 1)[0];
+    }
+
+    std::string provider_name = "";
+    bool provider_available = get_provider_name_from_settings (settings, provider_name);
```

Comment:
you probably want a default back and not a bool. This reads like a weird function and I am uncomfortable mutating inputs within reason. This feels like it should be:
`auto provider_name = get_provider_name(settings, provider_name);`

Reply:
settings is probably better named as `model_params`.

Reply:
done